tests/test_filters.py

```diff
@@ -871,6 +872,51 @@ def test_decimate_boundary():
     assert boundary.n_points
 
 
+def test_extract_surface():
+    # create a single hexahedral cell
+    lin_pts = np.array([[-1, -1, -1], # node 0
+                        [ 1, -1, -1], # node 1
+                        [ 1,  1, -1], # node 2
+                        [-1,  1, -1], # node 3
+                        [-1, -1,  1], # node 4
+                        [ 1, -1,  1], # node 5
+                        [ 1,  1,  1], # node 6
+                        [-1,  1,  1]], np.double) # node 7
+
+    quad_pts = np.array([
+        (lin_pts[1] + lin_pts[0])/2.0,
+        (lin_pts[1] + lin_pts[2])/2.0,
+        (lin_pts[2] + lin_pts[3])/2.0,
+        (lin_pts[3] + lin_pts[0])/2.0,
+        (lin_pts[4] + lin_pts[5])/2.0,
+        (lin_pts[5] + lin_pts[6])/2.0,
+        (lin_pts[6] + lin_pts[7])/2.0,
+        (lin_pts[7] + lin_pts[4])/2.0,
+        (lin_pts[0] + lin_pts[4])/2.0,
+        (lin_pts[1] + lin_pts[5])/2.0,
+        (lin_pts[2] + lin_pts[6])/2.0,
+        (lin_pts[3] + lin_pts[7])/2.0], np.double)
+
+    # introduce a minor variation to the location of the mid-side points
+    quad_pts += np.random.random(quad_pts.shape)*0.25
+    pts = np.vstack((lin_pts, quad_pts))
+
+    cells = np.asarray([[20, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19]], dtype=np.int64)
+    celltypes = np.array([VTK_QUADRATIC_HEXAHEDRON])
+    if pyvista._vtk.VTK9:
+        grid = pyvista.UnstructuredGrid(cells, celltypes, pts)
+    else:
+        grid = pyvista.UnstructuredGrid(np.array([0]), cells, celltypes, pts)
+
+    # expect each face to be divided 6 times since's it has a midside node
+    surf = grid.extract_surface()
+    assert surf.n_faces == 36
+
+    # expect each face to be divided several more times than the linear extraction
+    surf_subdivided = grid.extract_surface(subdivision=5)
+    assert surf_subdivided.n_faces > surf.n_faces
+
```

Comment:
Do we also want to test `subdivision=0`? In terms of code that would only test VTK, but we do specify this case in our docstring.

Reply:
I literally have 300 lines of code in a closed source project to trick my module to return this result.  I've added the test here and I'll refactor my internal code.  Learn something every day...